exchanges/coinbasepro/coinbasepro.go

```diff
@@ -3,761 +3,1297 @@ package coinbasepro
+	var params Params
+	params.Values = url.Values{}
+	err := params.encodeDateRange(startDate, endDate, startDateString, endDateString)
 	if err != nil {
-		return "", err
+		return nil, err
+	}
+	if contractExpiryType != "" {
+		params.Values.Set("contract_expiry_type", contractExpiryType)
 	}
+	if productType != "" {
+		params.Values.Set("product_type", productType)
+	}
+	if userNativeCurrency != "" {
+		params.Values.Set("user_native_currency", userNativeCurrency)
+	}
+	var resp TransactionSummary
+	return &resp, c.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, http.MethodGet, coinbaseV3+coinbaseTransactionSummary, params.Values, nil, true, &resp)
+}
 
-	return resp.ID, nil
+// CreateConvertQuote creates a quote for a conversion between two currencies. The trade_id returned can be used to commit the trade, but that must be done within 10 minutes of the quote's creation
+func (c *CoinbasePro) CreateConvertQuote(ctx context.Context, from, to, userIncentiveID, codeVal string, amount float64) (*ConvertResponse, error) {
+	if from == "" || to == "" {
+		return nil, errAccountIDEmpty
+	}
+	if amount == 0 {
+		return nil, order.ErrAmountIsInvalid
+	}
+	path := coinbaseV3 + coinbaseConvert + "/" + coinbaseQuote
+	tIM := map[string]any{
+		"user_incentive_id": userIncentiveID,
+		"code_val":          codeVal,
+	}
+	req := map[string]any{
+		"from_account":             from,
+		"to_account":               to,
+		"amount":                   strconv.FormatFloat(amount, 'f', -1, 64),
+		"trade_incentive_metadata": tIM,
+	}
+	resp := struct {
+		Trade ConvertResponse `json:"trade"`
+	}{}
+	return &resp.Trade, c.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, http.MethodPost, path, nil, req, true, &resp)
 }
 
-// PlaceMarketOrder places a new market order.
-// Orders can only be placed if the account has sufficient funds. Once an order
-// is placed, account funds will be put on hold for the duration of the order.
-// How much and which funds are put on hold depends on the order type and
-// parameters specified.
-//
-// GENERAL PARAMS
-// clientRef - [optional] Order ID selected by you to identify your order
-// side - 	buy or sell
-// productID - A valid product id
-// stp - [optional] Self-trade prevention flag
-//
-// MARKET ORDER PARAMS
-// size - [optional]* Desired amount in BTC
-// funds	[optional]* Desired amount of quote currency to use
-// * One of size or funds is required.
-func (c *CoinbasePro) PlaceMarketOrder(ctx context.Context, clientRef string, size, funds float64, side, productID, stp string) (string, error) {
-	resp := GeneralizedOrderResponse{}
-	req := make(map[string]any)
-	req["side"] = side
-	req["product_id"] = productID
-	req["type"] = order.Market.Lower()
+// CommitConvertTrade commits a conversion between two currencies, using the trade_id returned from CreateConvertQuote
+func (c *CoinbasePro) CommitConvertTrade(ctx context.Context, tradeID, from, to string) (*ConvertResponse, error) {
```

Comment:
Missing fields. No `is_netted`, so likely outdated struct definition. If I find one more, I'm going to ask you to double check all your work